include/deal.II/matrix_free/cuda_hanging_nodes_internal.h

```diff
@@ -787,6 +782,7 @@ namespace CUDAWrappers
           ((direction == 1) && ((constraint_mask & internal::constr_type_x) ?
                                   (x_idx == 0) :
                                   (x_idx == fe_degree)));
+        __syncthreads();
```

Comment:
I have to admit that it is not clear to me what could happen between this `__syncthreads` and the next one. It seems that we are only reading values into local variables... Is it possible that the synchronization problem happens before this function is called?

Reply:
I have not yet checked in detail what happens before, but couldn't the boolean operations (masking) that enable/disable certain lanes lead to some clash that manifests for some sizes, given that we have different threads (lanes) on different local dofs? That would at least be my interpretation, but then it has been a while since I last looked into how these boolean flags are translated into masked operations for the various lanes.

Reply:
@masterleinad Yes it is possible. Honestly I use Karl implementation as a reference and I was missing these `__synchtreads`. Adding them fix the problem but I am not sure why.